src/render/tutorials/caching-http-requests/2_0.html.md

```diff
@@ -0,0 +1,325 @@
+
+## Prerequisites
+- JDK 8
+- Docker
+
+## What you're going to learn:
+- How to wrap an existing (external) API with a caching proxy
+- How to use [behaviours](https://github.com/Knotx/knotx-fragments/tree/2.0.0/handler#behaviours) and [actions](https://github.com/Knotx/knotx-fragments/tree/2.0.0/handler#actions)
+
+Download the [Latest Knot.x Starter Kit release](https://github.com/Knotx/knotx-starter-kit/releases) and unzip it.
+
+Project has the following structure:
+```
+├── docker
+|   ├── Dockerfile                // Docker file with image definition.
+├── functional                    // Keep here your functional tests. Example implementation included
+├── gradle                        // Gradle wrapper and common gradle scripts
+├── knotx/conf                    // Knotx configuration which will be copied to docker image
+├── modules                       // Sub-modules of your project
+│   ├── ...                       // example modules implementation
+```
+
+## Actions & behaviours
+The first thing we are going to do, is define our custom task which will properly handle our request. As explained in the previous tutorials,
+Knot.x tasks are directed graphs of actions (if you want to get more familiar with configurable integrations please read [this](https://knotx.io/blog/configurable-integrations/) post). 
+If you want to learn more about tasks please check [knotx-fragments documentation](https://github.com/Knotx/knotx-fragments) 
+
+Let's start with defining a new task: `knotx/conf/routes/handlers/fragments-handler.conf`
+
+```hocon
+tasks {
+  get-product-task {
+    action = fetch-product-with-cache
+    onTransitions {
+      _success {
+        action = product-to-body
+      }
+    }
+  }
+}
+
+actions {
+  # Target API caching proxy
+  fetch-product-with-cache {
+    # https://github.com/Knotx/knotx-fragments/blob/2.0.0/handler/core/src/main/java/io/knotx/fragments/handler/action/InMemoryCacheActionFactory.java
+    factory = in-memory-cache
+    config {
+      cache {
+        maximumSize = 1000
+        ttl = 5000
+      }
+      cacheKey = "product-{param.id}"
+      payloadKey = fetch-product
+    }
+    doAction = fetch-product
+  }
+
+  # Target API
+  fetch-product {
+    # https://github.com/Knotx/knotx-data-bridge/blob/2.0.0/http/action/src/main/java/io/knotx/databridge/http/action/HttpActionFactory.java
+    factory = http
+    config {
+      endpointOptions {
+        path = /product/id
+        domain = webapi
+        port = 8080
+        allowedRequestHeaders = ["Content-Type"]
+      }
+    }
+
+  }
+
+  product-to-body {
+    # https://github.com/Knotx/knotx-fragments/blob/2.0.0/handler/core/src/main/java/io/knotx/fragments/handler/action/PayloadToBodyActionFactory.java
+    factory = payload-to-body
+    config {
+      key = fetch-product
+    }
+  }
+}
+```
+As you can see above, to achieve our goal we have to create only one task called `get-product-task` - this is very simple directed graph.
```

Comment:
Maybe:
>As you can see above, we declare the `get-product-task` task that points to the `fetch-product-with-cache` action node. The action declares one `success` transition (outgoing graph edge) that points to the `product-to-body` action node.

Reply:
You specify actions in the same paragraph, you do not have to mention that `you would describe them later`.

Reply:
fixed